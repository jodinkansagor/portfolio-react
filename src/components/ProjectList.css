.projectList {
  display: flex;
  flex-direction: column;
  align-items: center;

  ul {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 0px;
  }

  @media (min-width: 1200px) {
    ul {
      display: grid;
      grid-template-columns: 1fr 1fr 1fr;
      grid-gap: 7px;
    }
  }

}