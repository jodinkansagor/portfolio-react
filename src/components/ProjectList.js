import React from 'react';
import PropTypes from 'prop-types';
import styles from '../components/ProjectList.css';
import ProjectItem from '../components/ProjectItem';

//why is my linter unhappy here?

const ProjectList = ({ projectList }) => {
  const projectElements = projectList.map((projectItem) => (
    <li key={projectItem.name}>
      <ProjectItem {...projectItem} />
    </li>
  ));

  return (
    <div className={styles.projectList}>
      <h2>Select Projects</h2>
      <ul>
        {projectElements}
      </ul>
    </div>
  );
};

ProjectList.proptypes = {
  projectList: PropTypes.arrayOf(PropTypes.shape({
    img: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    stack: PropTypes.string.isRequired,
    siteLink: PropTypes.string.isRequired,
    githubLink: PropTypes.string.isRequired
  })).isRequired
};

export default ProjectList;
