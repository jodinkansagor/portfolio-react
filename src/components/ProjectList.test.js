import React from 'react';
import { shallow } from 'enzyme';
import ProjectList from './ProjectList';

describe('ProjectList component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<ProjectList projectList={['project', 'project']} />);
    expect(wrapper).toMatchSnapshot();
  });
});
