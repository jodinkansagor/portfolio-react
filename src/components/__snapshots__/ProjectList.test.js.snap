// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList component matches a snapshot 1`] = `
<div
  className="projectList"
>
  <h2>
    Select Projects
  </h2>
  <ul>
    <li
      key="0"
    >
      <ProjectItem
        0="p"
        1="r"
        2="o"
        3="j"
        4="e"
        5="c"
        6="t"
      />
    </li>
    <li
      key="1"
    >
      <ProjectItem
        0="p"
        1="r"
        2="o"
        3="j"
        4="e"
        5="c"
        6="t"
      />
    </li>
  </ul>
</div>
`;
